src/core/server/elasticsearch/client/client_config.ts

```diff
@@ -59,6 +59,10 @@ export function parseClientOptions(
     // do not make assumption on user-supplied data content
     // fixes https://github.com/elastic/kibana/issues/101944
     disablePrototypePoisoningProtection: true,
+    agent: {
+      maxSockets: Infinity,
```

Comment:
from https://github.com/elastic/kibana/issues/112756#issuecomment-932359607
```
When Kibana's communication with Elasticsearch occurs over TLS, there's rather significant overhead in establishing the TLS socket.
Therefore, if there's a stampeding herd of outbound requests to Elasticsearch we can end up in a situation where we try to create so many new TLS sockets at the same time that we end up getting TLS handshake timeouts.
In these situations, it can be beneficial to set maxSockets to a lower number, so that we don't try to establish so many new TLS sockets and we do HTTP request queueing in Kibana.
```
If we expect that `masxSockets: Inifinity` might affect performance for a TLS connection, 
maybe we can set a fixed limit for them? 
```js
agent: {
      maxSockets: config.ssl ? 512 : 1024
```

Reply:
This is why I suggested making `maxSockets` configurable, we don't know the optimal setting here.